### PR TITLE
Using github to create autocomplete inputs for generating github issues ...

### PR DESCRIPTION
**Pull request by ScottDowne**
...search urls

### CODE REVIEW

**Comment by ScottDowne** [comment]
There is some existing github api stuff in server.js that I didn't see until near the end. I can probably align with it.

**Comment by ScottDowne** [comment]
This isn't done yet.

**Comment by ScottDowne** [comment]
I think this is done enough to have review.

File: app/components/issues.jsx [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    this.getJson("/api/github/mozilla-milestones", function(data) {
+      this.setState({
+        milestones: data
+      });
+    }.bind(this));
+  },
+  render: function() {
+    var query = this.state.query;
+    var term = this.state.term;
+    var _this = this;
```
Comment (alicoding) [inline app/components/issues.jsx]: This is not being used anywhere.

File: app/components/issues.jsx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+              <option value="assignee" selected>Assignee</option>
+              <option value="author">Author</option>
+              <option value="label">Label</option>
+              <option value="milestone">Milestone</option>
+              <option value="repo">Repo</option>
+            </select>
+            <span className="input-container">
+              <input onChange={this.onChange} className="do-search-input" placeholder="search term" value={term} list={this.state.search}/>
+            </span>
+            <datalist id="assignee">
```
Comment (alicoding) [inline app/components/issues.jsx]: Can you do something like this instead?

```
var obj = {
  'assignee': 'user',
  'author': 'user'
  'repo' : 'repos'
};
var datalist = Object.keys(obj).map(item, index) {
  var id = Object.keys(obj[item]);
  return (<datalist id={item}>
               {this.state[id].map(function(item, index) {
                 return (<option key={index} value={item}>{item}</option>);
               })}
  </datalist>);
});
```

Untested code, but you get the idea.

File: server.js [hunk 3]
```diff
@@ -193,11 +189,132 @@ app.get('/dashboards', function(req, res) {
...
+    if (err) {
+      // Not sure what to do with errors, yet.
+      console.log(err);
+    } else {
+      res.json(results);
+    }
+  });
+});
+app.get( "/api/github/mozilla-labels", checkCache, function(req, res) {
+  var url = "http://127.0.0.1:" + app.get('port') + "/api/github/mozilla-repo-names";
```
Comment (alicoding) [inline server.js]: Hmm I'm not sure if that's a good idea to hard-coded  that url? `"http://127.0.0.1:"`?

File: server/models/github.js [hunk 4]
```diff
@@ -89,6 +88,42 @@ function Github(client, secret) {
...
+  var collection = [];
+
+  // Fetch deals with multiple pages.
+  // The data this code deals with is small enough,
+  // so just return all pages worth of data.
+  function fetch(page) {
+    request({
+      url: url + page,
+      headers: {
+        "user-agent": "build.webmaker.org" // GitHub is a diva without a unique user agent
```
Comment (alicoding) [inline server/models/github.js]: really? LOL

**Comment by alicoding** [comment]
I can't seem to test this locally since I can't even run the server or finish `npm install`... but I suspect that you need to update README and explain about Redis and why it's required now if that's what's this patch introduced.

File: server.js (hunk 3 above)
Comment (ScottDowne) [inline server.js]: It was advice from jbuck Should ping him if you have concerns and want it explained.

I didn't like it either.

Comment (alicoding) [inline server.js]: Can this instead be part of process.env or something that you need to define elsewhere?

Comment (ScottDowne) [inline server.js]: Doing some research on this, so far this is exactly my question: http://stackoverflow.com/questions/18516339/make-http-request-to-self-in-nodejs-expressjs

Comment (alicoding) [inline server.js]: I guess if this works locally and on prod then I think this is fine then.

File: app/components/issues.jsx (hunk 2 above)
Comment (ScottDowne) [inline app/components/issues.jsx]: Been meaning to make an AutoCompleteDataList React thingy instead, so I guess I'll do that now seeing how you're calling me on it.

File: server.js (hunk 3 above)
Comment (davidascher) [inline server.js]: Drive-by: why go through an HTTP call and not call the route directly?

Comment (ScottDowne) [inline server.js]: I found a better way, it wasn't sitting well with me either.

I _was_ doing it this way because the cache I was using was directly linked to the http call.

I ended up building the cache into API instead. I initially didn't want too much cache code mixed in with github code but I reduced it down to one spot which sits with me nicely. https://github.com/ScottDowne/build.webmaker.org/blob/a530713baed53700a9629083831d4427f86040a0/server/models/github.js#L95-L98

**Comment by ScottDowne** [comment]
@alicoding this is ready.

I had to make some changes to make it work on Heroku.

**Comment by alicoding** [comment]
Looking good. R+
